src/monster.cpp

```diff
@@ -1772,7 +1772,7 @@ int monster::heal( const int delta_hp, bool overheal )
     if( hp > maxhp && !overheal ) {
         hp = maxhp;
     }
-    return maxhp - old_hp;
+    return std::max( 0, hp - old_hp );
```

Comment:
The std::max is not needed. Health can not decrease.
This is due to delta_hp > 0 and hp < maxhp which is checked above.